nomad/structs/structs.go

```diff
@@ -2152,6 +2160,10 @@ const (
 	// TaskArtifactDownloadFailed indicates that downloading the artifacts
 	// failed.
 	TaskArtifactDownloadFailed = "Failed Artifact Download"
+
+	// TaskDiskExceeded indicates that one of the tasks in a taskgroup has
+	// exceeded the requested disk resources.
+	TaskDiskExceeded = "Disk Resources Exceeded"
```

Comment:
I'm not exactly sure what you want to do here. There is a message field on the TaskEvent, reason seems to indicate the same, maybe use a different name? Reason should be a(n embedded) struct to accommodate the two values?